python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -985,6 +987,32 @@ def fetch_row_counts(
                 A set of corresponding Dagster events for dbt models, with row counts attached,
                 yielded in the order they are emitted by dbt.
         """
+        return self._attach_metadata(_fetch_row_count_metadata)
+
+    def _attach_metadata(
```

Comment:
We're planning on making this a public method in the future, right? We should think of that name here. We don't have to add the `@public` decorator though. I think `map` could be a compelling name here. We'd need to make the fn type simpler: `Callable[DbtDagsterEventType, DbtDagsterEventType]`.

I can imagine this method to operate on modifying the general event stream, rather than just attaching metadata. This could be a pattern that exists outside of `dagster-dbt`.

Reply:
Right now the function is is useful as an internal simplification/util. If we want to expose it publicly, landing on the right name/signature is more tricky and warrants a more deliberate discussion. There are trade-offs either way - if we want to make it easy for users to easily attach metadata, we can more easily cache results or fold map calls together vs giving them power to map over all events. Either way I think it's beyond the scope of this PR and I don't feel that it's blocking what's here - we're not committing to anything name or signature wise.